### PR TITLE
Cloudwatch: Move getNextRefIdChar util from app/core/utils to @grafana/data

### LINKED ISSUE #80154: Remove core imports from `app/core/utils/query` from CloudWatch

### PR DESCRIPTION
**Pull request by idastambuk**
**What is this feature?**

Move getNextRefIdChar util from app/core/utils to @grafana/data

**Why do we need this feature?**

Part of https://github.com/grafana/grafana/issues/78778
As we didn't have an equivalent export from grafana/data and multiple plugins use this function from app/core, added a new util in grafana/data

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-data/src/query/refId.ts
Comment (sarahzinger) [inline packages/grafana-data/src/query/refId.ts:3]: I think for exported funcs in a library we should add some kind of comment explaining what the function can be used for. That way it can come up in intellisense. I also wonder if this is the best name for this func. Would it be more accurate to call it like "createNewRefId" or "getNextRefId" since at first I thought maybe this was returning a char of an existing refid, and on closer look it seems like it can return more than one char.

Also maybe we can go to wherever the original is and right a note to say that this is now an exported function

**Review by sarahzinger — COMMENTED** [review]
Looks neat a few small suggestions, but also defer to the original code owners of grafana-data on this!

File: packages/grafana-data/src/query/refId.ts
Comment (idastambuk) [reply packages/grafana-data/src/query/refId.ts]: Great suggestions, made the changes!

**Review by sarahzinger — APPROVED** [review]
Looks good to me! But maybe good to have someone from @grafana/grafana-frontend-platform or @grafana/plugins-platform-frontend give a look too if possible

File: public/app/core/utils/query.ts
Comment (Clarity-89) [inline public/app/core/utils/query.ts:3]:
```suggestion
// @deprecared use the `getNextRefId` function from grafana/data instead
```
I think if you change the comment to this, the code editors (at least the IntelliJ-based ones) will mark this function as deprecated.

**Review by Clarity-89 — APPROVED** [review]
Looks good, left a minor comment.

File: public/app/core/utils/query.ts
Comment (idastambuk) [reply public/app/core/utils/query.ts]: Done, thanks @Clarity-89!

**Pull request merged**
